packages/cli/src/util/paths.ts

```diff
@@ -0,0 +1,168 @@
+import { generate } from '@stoplight/prism-http/src/mocker/generator/HttpParamGenerator';
+import { serializeWithDeepObjectStyle } from '@stoplight/prism-http/src/mocker/serializer/style/deepObject';
+import {
+  serializeWithPipeDelimitedStyle,
+  serializeWithSpaceDelimitedStyle,
+} from '@stoplight/prism-http/src/mocker/serializer/style/delimited';
+import {
+  Dictionary,
+  HttpParamStyles,
+  IHttpOperation,
+  IHttpParam,
+  IHttpPathParam,
+  IHttpQueryParam,
+  INodeExample,
+  INodeExternalExample,
+} from '@stoplight/types';
+import * as Array from 'fp-ts/lib/Array';
+import * as Either from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get } from 'lodash';
+// @ts-ignore
+import { parse } from 'uri-template';
+
+export function createExamplePath(operation: IHttpOperation): Either.Either<Error, string> {
+  return pipe(
+    generateTemplateAndValuesForPathParams(operation),
+    Either.chain(({ template: pTemplate, values: pValues }) => {
+      return pipe(
+        generateTemplateAndValuesForQueryParams(pTemplate, operation),
+        Either.map(({ template: qTemplate, values: qValues }) => {
+          return { template: qTemplate, values: { ...pValues, ...qValues } };
+        }),
+      );
+    }),
+    Either.map(({ template, values }) => parse(template).expand(values)),
+  );
+}
+
+function generateParamValue(spec: IHttpParam): Either.Either<Error, unknown> {
+  return pipe(
+    generate(spec),
+    Either.fromOption(() => new Error(`Cannot generate value for: ${spec.name}`)),
+    Either.chain(value => {
+      switch (spec.style) {
+        case HttpParamStyles.DeepObject:
+          return serializeWithDeepObjectStyle(spec.name, value);
+
+        case HttpParamStyles.PipeDelimited:
+          return serializeWithPipeDelimitedStyle(spec.name, value as Array<string | number | boolean>, spec.explode);
```

Comment:
I think _this_ is the place where we should do the `Array.isArray` and return an error in case it's not working, not in the function; so we can

1. Eliminate the cast
2. Make the internal function care about its job and not having to validate the arguments. Does it make sense @StefanDywersant?

Reply:
Yeah, you completely right. Checking for it inside the serializer was a bad decision. Fixed.